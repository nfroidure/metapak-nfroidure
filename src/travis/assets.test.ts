import { describe, beforeEach, jest, test, expect } from '@jest/globals';
import _fs from 'fs';
import path from 'path';
import assetsTransformer from './assets.js';
import type { LogService, FSService } from 'metapak';

describe('TravisCI', () => {
  const fs = {
    readFileAsync: jest.fn<FSService['readFileAsync']>(),
    writeFileAsync: jest.fn<FSService['writeFileAsync']>(),
    mkdirpAsync: jest.fn<FSService['mkdirpAsync']>(),
    accessAsync: jest.fn<FSService['accessAsync']>(),
    readdirAsync: jest.fn<FSService['readdirAsync']>(),
    unlinkAsync: jest.fn<FSService['unlinkAsync']>(),
  };
  const PROJECT_DIR = '/lol/';
  const log = jest.fn<LogService>();

  beforeEach(() => {
    fs.readFileAsync.mockReset();
    fs.writeFileAsync.mockReset();
    fs.mkdirpAsync.mockReset();
    fs.accessAsync.mockReset();
    fs.readdirAsync.mockReset();
    fs.unlinkAsync.mockReset();
    log.mockReset();
  });

  describe('Assets transformer', () => {
    test('should fill the manifest', async () => {
      expect(
        await assetsTransformer(
          {
            name: '.travis.yml',
            data: _fs.readFileSync(
              path.join('src', 'travis', 'assets', '_dot_travis.yml'),
              'utf-8',
            ),

            dir: '/home/whoami/project/dir',
          },
          {
            metapak: {
              configs: ['www'],
              data: {},
            },
          },
          {
            PROJECT_DIR,
            fs,
            log,
          },
        ),
      ).toMatchInlineSnapshot(`
        {
          "data": "# This file is automatically generated by a \`metapak\`
        # module. Do not change it elsewhere, changes would
        # be overridden.

        language: node_js
        node_js:
          - 16
          - 16.15.0
          - 17
          - 18
        ",
          "dir": "/home/whoami/project/dir",
          "name": ".travis.yml",
        }
      `);
    });

    test('should let pass other files', async () => {
      expect(
        await assetsTransformer(
          {
            name: 'YOLO',
            data: 'Carpe diem\n',
            dir: '/home/whoami/project/dir',
          },
          {
            metapak: {
              configs: ['www'],
              data: {},
            },
          },
          {
            PROJECT_DIR,
            fs,
            log,
          },
        ),
      ).toMatchInlineSnapshot(`
        {
          "data": "Carpe diem
        ",
          "dir": "/home/whoami/project/dir",
          "name": "YOLO",
        }
      `);
    });
  });
});
