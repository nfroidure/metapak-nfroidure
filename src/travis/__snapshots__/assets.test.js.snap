// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TravisCI Assets transformer should fill the manifest 1`] = `
Object {
  "data": "# This file is automatically generated by a \`metapak\`
# module. Do not change it elsewhere, changes would
# be overridden.

language: node_js
node_js:
  - 6
  - 6.9.5
  - 7
  - 8
  - 9
",
  "name": ".travis.yml",
}
`;
